eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1840,18 +1850,44 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
     }
   }
 
-  def handleCommandSuccess(sender: ActorRef, newData: Data) = {
-    stay using newData replying ChannelCommandResponse.Ok
+  def handleFastClose(c: CloseCommand, channelId: ByteVector32) = {
+    goto(CLOSED) replying RES_SUCCESS(c, channelId)
+  }
+
+  def handleCommandSuccess(c: Command, newData: Data) = {
+    val replyTo = c match {
+      case hasReplyTo: HasReplyTo if hasReplyTo.replyTo != ActorRef.noSender => hasReplyTo.replyTo
+      case _ => sender
+    }
+    val channelId = Helpers.getChannelId(newData)
+    replyTo ! RES_SUCCESS(c, channelId)
+    stay using newData
+  }
+
+  def handleAddHtlcCommandError(c: CMD_ADD_HTLC, cause: Throwable, channelUpdate: Option[ChannelUpdate]) = {
```

Comment:
True, although it also would break the symmetry with other `Commitments` functions. For some operations (e.g. signature) we can't restrict to known `ChannelException`. Hmm... should we encapsulate those is a `BitcoinlibException`?

Reply:
Good question...probably for later though, even if we had such a `BitcoinlibException` I think it's still useful that every method returns the "smallest" type it can, so `sendAdd` could still return only `ChannelExceptions`

Reply:
That's sounds reasonable, I'll look into it.

Reply:
Done with a0ae9f81e.